src/error.rs

```diff
@@ -85,6 +85,19 @@ impl core::fmt::Display for StatusCode {
     }
 }
 
+impl StatusCode {
+    /// If the ffi status is unknown/undefined, use not_supported by default.
+    /// MsQuic returns an undefined status code should not happen ideally,
```

Comment:
MsQuic absolutely can return other stays codes from the platform. I think we need a better solution for that.

Reply:
Removed this. MsQuic returned code will use the Status struct and it can represent arbitrary numbers.
When user wants to check the StatusCode enum, they need to call Status::try_as_status_code, and this enum conversion might return failure if the code is not defined in QUIC_STATUS enum, but some other platform error code.